catalyst-gateway/bin/src/service/api/cardano/rbac/mod.rs

```diff
@@ -1,5 +1,56 @@
 //! RBAC endpoints.
 
-pub(crate) mod chain_root_get;
-pub(crate) mod registrations_get;
-pub(crate) mod role0_chain_root_get;
+use poem_openapi::{param::Query, payload::Json, OpenApi};
+
+use crate::service::common::{
+    auth::{none_or_api_key::NoneOrApiKey, none_or_rbac::NoneOrRBAC},
+    tags::ApiTags,
+    types::cardano::query::stake_or_voter::StakeOrVoter,
+};
+
+mod rbac_registration;
+mod registrations_get;
+mod unprocessable_content;
+
+/// Cardano RBAC API Endpoints
+pub(crate) struct Api;
+
+#[OpenApi(tag = "ApiTags::Cardano")]
+impl Api {
+    #[oai(
+        path = "/draft/rbac/registrations/:catalyst_id",
```

Comment:
catalyst ID can not be in the path, its a URI. 
```suggestion
        path = "/draft/rbac/registration",
```